### ISSUE TITLE
Bug: accept4: too many open files

### ISSUE THREAD

**Issue by Poeschl** [containous/traefik#3933]
Unfortunately, the `to many open files` of #1322 is still a thing.
Tonight the traefik on a prod environment showed us tones of this log:

```
time="2018-09-25T23:04:33Z" level=debug msg="http: Accept error: accept tcp [::]:443: accept4: too many open files; retrying in 5ms"
time="2018-09-25T23:04:33Z" level=debug msg="http: Accept error: accept tcp [::]:443: accept4: too many open files; retrying in 10ms"
time="2018-09-25T23:04:33Z" level=debug msg="http: Accept error: accept tcp [::]:443: accept4: too many open files; retrying in 20ms"
time="2018-09-25T23:04:33Z" level=debug msg="http: Accept error: accept tcp [::]:443: accept4: too many open files; retrying in 40ms"
time="2018-09-25T23:04:33Z" level=debug msg="http: Accept error: accept tcp [::]:443: accept4: too many open files; retrying in 80ms"
time="2018-09-25T23:04:33Z" level=debug msg="http: Accept error: accept tcp [::]:443: accept4: too many open files; retrying in 160ms"
time="2018-09-25T23:04:33Z" level=debug msg="http: Accept error: accept tcp [::]:443: accept4: too many open files; retrying in 320ms"
time="2018-09-25T23:04:33Z" level=debug msg="http: Accept error: accept tcp [::]:443: accept4: too many open files; retrying in 640ms"
time="2018-09-25T23:04:34Z" level=debug msg="http: Accept error: accept tcp [::]:443: accept4: too many open files; retrying in 1s"
time="2018-09-25T23:04:35Z" level=debug msg="http: Accept error: accept tcp [::]:443: accept4: too many open files; retrying in 1s"
time="2018-09-25T23:04:36Z" level=debug msg="http: Accept error: accept tcp [::]:443: accept4: too many open files; retrying in 1s"
time="2018-09-25T23:04:37Z" level=debug msg="http: Accept error: accept tcp [::]:443: accept4: too many open files; retrying in 1s"
time="2018-09-25T23:04:38Z" level=debug msg="http: Accept error: accept tcp [::]:443: accept4: too many open files; retrying in 1s"
time="2018-09-25T23:04:39Z" level=debug msg="http: Accept error: accept tcp [::]:443: accept4: too many open files; retrying in 1s"
time="2018-09-25T23:04:40Z" level=debug msg="http: Accept error: accept tcp [::]:443: accept4: too many open files; retrying in 1s"
time="2018-09-25T23:04:41Z" level=debug msg="http: Accept error: accept tcp [::]:443: accept4: too many open files; retrying in 1s"
time="2018-09-25T23:04:42Z" level=debug msg="http: Accept error: accept tcp [::]:443: accept4: too many open files; retrying in 1s"
```
Until the container runs out of its memory.
At that time we had just a low load with around 85 Requests/min.

![image](https://user-images.githubusercontent.com/5469257/46063571-86aed180-c16d-11e8-929c-b8ab850088b5.png)
The red line is the RAM soft-limit of 32MB

### What version of Traefik are you using (traefik version)?
version: we use the official `traefik:1.6.4-alpine` docker image.

### What is your environment & configuration (arguments, toml...)?

```toml
# traefik.toml
logLevel = "DEBUG"
#traefikLogsFile = "/log/traefik.log"
defaultEntryPoints = ["http", "https"]

[accessLog]

# Web configuration backend
[web]
address = ":8080"
[web.auth.basic]
users = ["admin:$2a$12$E8bd5WdgnNxGFo6z3k/zV.zz8In9/Oa/mBDf/p4oHDWcl.2sydaU6"]

# Docker configuration backend
[docker]
watch = true
exposedbydefault = false

#Entrypoints
[entryPoints]
  [entryPoints.http]
  address = ":80"
    [entryPoints.http.redirect]
      entryPoint = "https"
  [entryPoints.https]
  address = ":443"
    [entryPoints.https.tls]
      [[entryPoints.https.tls.certificates]]
      CertFile = "/etc/traefik/cert.pem"
      KeyFile = "/etc/traefik/key.unencrypted.pem"

[acme]
email = "******@pixel.de"
storage = "/etc/traefik/acme.json"
entryPoint = "https"
#onDemand = true
acmeLogging = true
#OnHostRule = true

[acme.httpChallenge]
  entrypoint = "http"

```
This config contains some obfuscated data and the backends are configured inside of their docker config.

**Comment by LoicAG** [comment]
Same issue here, and given the lack of support, we might end up giving up on traefik entirely.

**Comment by ldez** [comment]
@LoicAG you can help us by providing a reproducible case.

You can also have a [commercial support](https://traefik.io/) if the community support ([Github](https://github.com/containous/traefik/issues), [Slack](https://slack.traefik.io/), [Stackoverflow](https://stackoverflow.com/questions/tagged/traefik)) is not enough.

**Comment by IlariExove** [comment]
I think we're hitting this on Traefik 2.0.2.

`time="2019-10-16T12:55:34Z" level=error msg="accept tcp [::]:4430: accept4: too many open files" entryPointName=web-secure`

And then no more entries in error log and it stops accepting new connections.

**Comment by dduportal** [comment]
Hi @IlariExove , given the age of this issue (Traefik v1.6 ...), do you mind opening a new issue with reproduction elements: your Traefik configuration, the kernel configuration of the machine where you have the problem, soft and hard limits configured on this machine, capabilities of the machine, metrics about the number of TCP and open files, and any other useful information to help us analyse this behavior. Thanks a lot!

**Comment by eisenwinter** [comment]
we faced this issue after moving more traffic trough v2, while it worked fine for weeks with merly traffik of course, we increased the file descriptor limit as a workarround for now (ulimit -n <SIZE>) wich seemed to tackle the issue for now but I guess this isnt really a long-term solution considering v1 ran on the same settings for years without problems dealing with far more traffic

**Comment by IlariExove** [comment]
In our case the problem was most likely in our iptables rules with REDIRECT actions in nat tables. We simplified the network setup and the problem vanished.

**Comment by IlariExove** [comment]
But I'm thinking that a feature request ticket for v2 mayhaps could be opened to improve handling of "too many open files" error. The current implementation doesn't recover by itself and the only way to restore traffic is to restart Traefik. The most simplest way to handle the exception is to make Traefik restart by itself.

**Comment by traefiker** [comment]
Closed by #5935.
